src/App.tsx

```diff
@@ -14,11 +15,115 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isReversed: boolean,
+  sortBy: string,
+  appLoad: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isReversed: false,
+    sortBy: '',
+    appLoad: false,
+  };
+
+  start = () => {
+    this.setState({ appLoad: true });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  sortByLength = () => {
+    this.reset();
+    this.setState({ sortBy: 'length' });
+  };
```

Comment:
you two times call setState: here and on reset. Maybe should do this just once?)